tests/e2e/svmotion_volumes.go

```diff
@@ -0,0 +1,1348 @@
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+/*
+	Test to verify sVmotion works fine for volumes in detached state
+
+	Steps
+	1. Create StorageClass.
+	2. Create PVC.
+	3. Expect PVC to pass and verified it is created correctly.
+	4. Relocate detached volume
+	5. Invoke CNS Query API and validate datastore URL value changed correctly
+*/
+
+var _ = ginkgo.Describe("[csi-block-vanilla] [csi-block-vanilla-parallelized] Relocate detached volume ", func() {
+	f := framework.NewDefaultFramework("svmotion-disk")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	var (
+		client          clientset.Interface
+		namespace       string
+		scParameters    map[string]string
+		datastoreURL    string
+		sourceDatastore *object.Datastore
+		destDatastore   *object.Datastore
+		datacenter      *object.Datacenter
+		destDsURL       string
+		pvclaims        []*v1.PersistentVolumeClaim
+		fcdID           string
+		labelKey        string
+		labelValue      string
+	)
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		scParameters = make(map[string]string)
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		destDsURL = GetAndExpectStringEnvVar(destinationDatastoreURL)
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		labelKey = "app"
+		labelValue = "e2e-labels"
+	})
+
+	// Test for relocating volume being detached state
+	ginkgo.It("Verify relocating detached volume works fine", func() {
+		ginkgo.By("Invoking Test for relocating detached volume")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		finder := find.NewFinder(e2eVSphere.Client.Client, false)
+		cfg, err := getConfig()
+		var datacenters []string
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		dcList := strings.Split(cfg.Global.Datacenters,
+			",")
+		for _, dc := range dcList {
+			dcName := strings.TrimSpace(dc)
+			if dcName != "" {
+				datacenters = append(datacenters, dcName)
+			}
+		}
+
+		for _, dc := range datacenters {
+			datacenter, err = finder.Datacenter(ctx, dc)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			finder.SetDatacenter(datacenter)
+			sourceDatastore, err = getDatastoreByURL(ctx, datastoreURL, datacenter)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			destDatastore, err = getDatastoreByURL(ctx, destDsURL, datacenter)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		scParameters[scParamDatastoreURL] = datastoreURL
+		storageclass, pvclaim, err := createPVCAndStorageClass(client,
+			namespace, nil, scParameters, "", nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
```

Comment:
Add log messages to all the major steps

Reply:
Done

Please note that this test is not part of my changes(this is already present on master), I just renamed -> svmotion_volumes.go, hence it is coming as part of my changes.